Fix tests for both python2.7 and python3.6, drop python2.6

gyst opened the pull request.

gyst committed:
Update for python3.6:
- provide requirements.txt to bypass bootstrap.py using pip
- avoid 'python' interpreter name clash in virtualenv
- add python3.6 to travis
- add some gitignores

gyst committed:
Tickle Travis

gyst committed:
Fix tests for python2.7, drop python2.6, update trove classifiers

janwijbrand commented:
Before we can merge we need:

* a tox.ini 
* update travis.yml to include all the supported python versions
* update setup.py's classifiers accordingly
* remove requirements.txt
* invoke the tests to the established pattern

gyst committed:
Avoid buildout for tests.
- Use pip for travis
- Provide tox.ini, using same invocations,
- Support py3.4, py3.5, pypy, pypy3,
- Update trove classifiers

gyst commented:
I've done all of that, except removing `requirements.txt`. As long as we have `buildout.cfg`, `requirements.txt` provides an improvement over `bootstrap.py`.

gyst committed:
Add coverage to tox, fixup buildout
